inginious/frontend/submission_manager.py

```diff
@@ -49,11 +49,21 @@ def _job_done_callback(self, submissionid, task, result, grade, problems, tests,
         submission = self.get_submission(submissionid, False)
         submission = self.get_input_from_submission(submission)
 
+        # Get user penalty stored in 'user_hints' collection to apply in grade of submission
+        # TODO: should be done for submissions in group tasks mode
+        penalty = 0.0
+        if grade:
+            penalty = \
+                self._hook_manager.call_hook('show_hints', taskid=task.get_id(), username=submission["username"][0],
+                                             database=self._database)[0]
```

Comment:
You call this hook which calls the function `get_user_total_penalty`, but that function only returns a number, not an array. Why do you have this? I got an exception.

Reply:
This needs to be fixed before I merge this. Please do this.